assets/app/scripts/controllers/overview.js

```diff
@@ -384,7 +416,114 @@ angular.module('openshiftConsole')
       });
     });
 
+    /*
+     * The topology graph wants unique identifiers.
+     *
+     * One of the warts on the kubernetes/openshift API is that there
+     * is no unique global identifier per resource. We construct them here from
+     * Kind/metadata.name ... which is good enough in a single namespace.
+     */
+    function makeId(name, kind) {
```

Comment:
uid is actually globally unique... an openshift project *is* a namespace, and an openshift deployment *is* a replication controller